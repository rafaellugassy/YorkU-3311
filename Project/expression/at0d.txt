addition

counting

start_of_set_enumeration
bool_value(true)
bool_value(false)
bool_value(true)
bool_value(false)
end_of_set_enumeration

counting

start_of_set_enumeration
greater_than
int_value(4)
int_value(3)
greater_than
int_value(3)
int_value(4)
less_than
int_value(0)
int_value(1)
end_of_set_enumeration

analyze
simplify
