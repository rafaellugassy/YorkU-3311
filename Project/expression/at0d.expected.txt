  Expression currently specified: ?
  Report: Expression is initialized.
->addition
  Expression currently specified: (? + nil)
  Report: OK.
->counting
  Expression currently specified: ((# ?) + nil)
  Report: OK.
->start_of_set_enumeration
  Expression currently specified: ((# {?}) + nil)
  Report: OK.
->bool_value(True)
  Expression currently specified: ((# {True, ?}) + nil)
  Report: OK.
->bool_value(False)
  Expression currently specified: ((# {True, False, ?}) + nil)
  Report: OK.
->bool_value(True)
  Expression currently specified: ((# {True, False, True, ?}) + nil)
  Report: OK.
->bool_value(False)
  Expression currently specified: ((# {True, False, True, False, ?}) + nil)
  Report: OK.
->end_of_set_enumeration
  Expression currently specified: ((# {True, False, True, False}) + ?)
  Report: OK.
->counting
  Expression currently specified: ((# {True, False, True, False}) + (# ?))
  Report: OK.
->start_of_set_enumeration
  Expression currently specified: ((# {True, False, True, False}) + (# {?}))
  Report: OK.
->greater_than
  Expression currently specified: ((# {True, False, True, False}) + (# {(? > nil)}))
  Report: OK.
->int_value(4)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > ?)}))
  Report: OK.
->int_value(3)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), ?}))
  Report: OK.
->greater_than
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (? > nil)}))
  Report: OK.
->int_value(3)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > ?)}))
  Report: OK.
->int_value(4)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), ?}))
  Report: OK.
->less_than
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (? < nil)}))
  Report: OK.
->int_value(0)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < ?)}))
  Report: OK.
->int_value(1)
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < 1), ?}))
  Report: OK.
->end_of_set_enumeration
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < 1)}))
  Report: OK.
->analyze
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < 1)}))
  Report: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < 1)})) is type-correct.
->simplify
  Expression currently specified: ((# {True, False, True, False}) + (# {(4 > 3), (3 > 4), (0 < 1)}))
  Report: 4
